packages/clerk-js/src/utils/authPropHelpers.ts

```diff
@@ -1,48 +1,48 @@
 import { camelToSnake } from '@clerk/shared';
-import type { DisplayConfigResource } from '@clerk/types';
+import type { ClerkOptions, DisplayConfigResource } from '@clerk/types';
 import type { ParsedQs } from 'qs';
 import qs from 'qs';
 
-import { hasBannedProtocol, isValidUrl } from '../../utils';
-import type { SignInCtx, SignUpCtx } from '../types';
+import { hasBannedProtocol, isValidUrl } from './index';
 
-type ExtractAuthUrlKey =
-  | keyof Pick<SignUpCtx, 'afterSignUpUrl' | 'afterSignInUrl'>
-  | keyof Pick<SignInCtx, 'afterSignUpUrl' | 'afterSignInUrl'>;
+type ExtractAuthUrlKey = 'afterSignUpUrl' | 'afterSignInUrl' | 'signInUrl' | 'signUpUrl';
 
-type ExtractAuthPropOptions =
-  | { queryParams: ParsedQs; displayConfig: DisplayConfigResource } & (
-      | {
-          ctx: Omit<SignUpCtx, 'componentName'>;
-        }
-      | {
-          ctx: Omit<SignInCtx, 'componentName'>;
-        }
-    );
+type ExtractAuthOptions = {
+  queryParams?: ParsedQs;
+  displayConfig?: DisplayConfigResource;
+  options?: ClerkOptions;
+  ctx?: any;
+};
 
 /**
  *
- * extractAuthProp(key, options)
+ * extractAuthPathByPriority(key, options, accessRedirectUrl)
  *
- * Retrieve auth redirect props passed through querystring parameters, component props
+ * Retrieve auth redirect props passed through querystring parameters, component props, ClerkProvider props
  * or display config settings.
  *
  * The priority is:
- * Querystring parameters > ClerkJS component props > Display configuration payload
+ * 1. Querystring parameters
+ * 2. ClerkJS component props
+ * 3. ClerkProvider props
+ * 4. Display configuration payload
  */
-export const extractAuthProp = (
+export const extractAuthPathByPriority = (
   key: ExtractAuthUrlKey,
-  { ctx, queryParams, displayConfig }: ExtractAuthPropOptions,
+  { ctx, queryParams, displayConfig, options }: ExtractAuthOptions,
+  accessRedirectUrl = true,
 ): string => {
   const snakeCaseField = camelToSnake(key);
-  const queryParamValue = queryParams[snakeCaseField];
+  const queryParamValue = queryParams?.[snakeCaseField];
 
   const url =
     (typeof queryParamValue === 'string' ? queryParamValue : null) ||
-    (typeof queryParams.redirect_url === 'string' ? queryParams.redirect_url : null) ||
-    ctx[key] ||
-    ctx.redirectUrl ||
-    displayConfig[key];
+    (accessRedirectUrl && typeof queryParams?.redirect_url === 'string' ? queryParams?.redirect_url : null) ||
+    ctx?.[key] ||
+    (accessRedirectUrl ? ctx?.redirectUrl : null) ||
+    options?.[key] ||
+    displayConfig?.[key] ||
+    undefined;
```

Comment:
🙃 why do we need `undefined` as default ? Also what do you think about replacing the ternary operator conditions with `&&` ? eg:
```
(
   (accessRedirectUrl && queryParams?.redirect_url) || 
   ctx?.[key] ||
   (accessRedirectUrl && ctx?.redirectUrl) ||
   options?.[key] ||
   displayConfig?.[key] ||
   undefined
 )

Reply:
@dimkl Do you think replacing with `&&` will be easier to read?

Reply:
(personal opinion) yes but leave it as is?